gitflow_toolbox/common/gitlab.py

```diff
@@ -0,0 +1,55 @@
+import os
+
+import gitlab
+from gitlab.v4.objects.projects import Project
+
+from gitflow_toolbox.common.get_env import get_env
+from gitflow_toolbox.common.singleton import Singleton
+
+
+class CurrentGitlab(gitlab.Gitlab, metaclass=Singleton):
+
+    current_project_memo = None
+
+    def __init__(self):
+        # If running in a gitlab CI job
+        if os.getenv("GITLAB_CI"):
+            # Use provided token, if not available, use default token from job
+            private_token = get_env("GITLAB_PRIVATE_TOKEN", "CI_JOB_TOKEN")
+            super().__init__(get_env("CI_SERVER_URL"), private_token)
+        else:
+            super().__init__(get_env("GITLAB_URL"), get_env("GITLAB_PRIVATE_TOKEN"))
+
+    def current_project(self) -> Project:
```

Comment:
We already know that it's the current or the remote project as it's indicated in the class name.

Should be renamed "project"
If it's not a function with params, it could be an accessor

```python
@property
def project(self) -> Project:
        """Get current gitlab sdk project

        Returns:
            gitlab.Gitlab: current gitlab project instance (or 'source', 'from' gitlab project)
        """
        if not self.current_project_memo:
            # If running in a gitlab CI job
            if os.getenv("GITLAB_CI"):
                # Use provided CI project, if not available, use project from current job
                project_id = get_env("GITLAB_PROJECT_ID", "CI_PROJECT_ID")
                self.current_project_memo = self.projects.get(project_id)

            self.current_project_memo = self.projects.get(get_env("GITLAB_PROJECT_ID"))
        return self.current_project_memo
```

Same thing, but we don't need to call it

Reply:
I was anticipating eventual future developments where there can be multiple local repo and multiple remote repos, my design will not produce breaking changes. But for now, with the env vars I set, you're right.

Reply:
Hum, I can agree with that.

In the future, we will need to remove this env vars computation to support multiple current / remote repo / projets